ui/admin/app/routes/scopes/scope/aliases/index.js

```diff
@@ -4,10 +4,100 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
 
 export default class ScopesScopeAliasesIndexRoute extends Route {
+  // =services
+
+  @service store;
+  @service session;
+  @service can;
+  @service router;
+
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
   // =methods
 
+  /**
+   * Loads queried aliases and the number of aliases under current scope.
+   * @returns {Promise<{totalItems: number, aliases: [object], aliasesExist: boolean}>}
+   */
+  async model({ search, page, pageSize }) {
+    const scope = this.modelFor('scopes.scope');
+    const { id: scope_id } = scope;
+
+    let aliases;
+    let aliasesExist;
+    let totalItems = 0;
+
+    if (
+      this.can.can('list model', scope, {
+        collection: 'aliases',
+      })
+    ) {
+      aliases = await this.store.query('alias', {
+        scope_id,
+        query: { search },
+        page,
+        pageSize,
+      });
+      totalItems = aliases.meta?.totalItems;
+      // since we don't receive target info from aliases list API,
+      // we query the store to fetch target information based on the destination id
+      aliases = await Promise.all(
+        aliases.map((alias) =>
+          hash({
+            alias,
+            target: alias.destination_id
+              ? this.store.findRecord('target', alias.destination_id)
```

Comment:
Do you think we should set `backgroundReload` to false to help prevent extra queries if it already exists in the store? I imagine we shouldn't need to try and refresh data that was already in the store, especially if people are going between pages

Reply:
good point, i didnt realize that was an option, originally thought that not includeing `reload: true` was sufficient to prevent extra queries.